aspnetcore/tutorials/first-mvc-app/adding-model.md

```diff
@@ -20,7 +20,393 @@ The model classes you create are known as POCO classes (from **P**lain **O**ld *
+
+Create a *Data* folder.
+
+Add a *Data/MvcMovieContext.cs* file with the following code: 
+
+[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/zDocOnly/MvcMovieContext.cs?name=snippet)]
+
+The preceding code creates a [DbSet\<Movie>](/dotnet/api/microsoft.entityframeworkcore.dbset-1) property for the entity set. In Entity Framework terminology, an entity set typically corresponds to a database table. An entity corresponds to a row in the table.
+
+<a name="reg"></a>
+
+## Register the database context
+
+ASP.NET Core is built with [dependency injection (DI)](xref:fundamentals/dependency-injection). Services (such as the EF Core DB context) must be registered with DI during application startup. Components that require these services (such as Razor Pages) are provided these services via constructor parameters. The constructor code that gets a DB context instance is shown later in the tutorial. In this section, you register the database context with the DI container.
+
+Add the following `using` statements at the top of *Startup.cs*:
+
+```csharp
+using MvcMovie.Data;
+using Microsoft.EntityFrameworkCore;
+```
+
+Add the following highlighted code in `Startup.ConfigureServices`:
+
+# [Visual Studio](#tab/visual-studio)
+
+[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/Startup.cs?name=snippet_ConfigureServices&highlight=6-7)]
+
+# [Visual Studio Code / Visual Studio for Mac](#tab/visual-studio-code+visual-studio-mac)
+
+[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/Startup.cs?name=snippet_UseSqlite&highlight=6-7)]
+
+---
+
+The name of the connection string is passed in to the context by calling a method on a [DbContextOptions](/dotnet/api/microsoft.entityframeworkcore.dbcontextoptions) object. For local development, the [ASP.NET Core configuration system](xref:fundamentals/configuration/index) reads the connection string from the *appsettings.json* file.
+
+<a name="cs"></a>
+
+## Add a database connection string
+
+Add a connection string to the *appsettings.json* file:
+
+# [Visual Studio](#tab/visual-studio)
+
+[!code-json[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/appsettings.json?highlight=10-12)]
+
+# [Visual Studio Code / Visual Studio for Mac](#tab/visual-studio-code+visual-studio-mac)
+
+[!code-json[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/appsettings_SQLite.json?highlight=10-12)]
+
+---
+
+Build the project as a check for compiler errors.
+
+## Scaffold movie pages
+
+Use the scaffolding tool to produce Create, Read, Update, and Delete (CRUD) pages for the movie model.
+
+# [Visual Studio](#tab/visual-studio)
+
+In **Solution Explorer**, right-click the *Controllers* folder **> Add > New Scaffolded Item**.
```

Comment:
> In order for scaffolding to run we need to make sure the audience has the following package added first:
> Microsoft.VisualStudio.Web.CodeGenerators.Mvc" Version="5.0.0-rc.2.20516.1"
> 
> Otherwise they will get this error:
> ![image](https://user-images.githubusercontent.com/10985336/98498116-d2f8e680-21fa-11eb-90c4-71e4ce70e202.png)

The images were correct. I sent you links to show that. For some reason it kept rendering the incorrect images. I made copies of each year old image and reference the copy image, now it's rendering correctly.  Not sure how it could cache an image more than a year old.